### PR TITLE
Upgrade to latest Docker

### PR DESCRIPTION
**Pull request by jfalcou**
g++ 12 is on

### CODE REVIEW

File: cmake/toolchain/random.targets.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "random.exe": "^random\\..*\\.exe"
```
Comment (DenisYaroshevskiy) [inline cmake/toolchain/random.targets.json:2]: What's this?

File: .github/actions/integration/entrypoint.sh [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #!/bin/sh -l
 set -e
 
+# Setup github/worspace as safe
+git config --global --add safe.directory /github/workspace
```
Comment (DenisYaroshevskiy) [inline .github/actions/integration/entrypoint.sh:5]: What's this?

**Review by DenisYaroshevskiy — APPROVED** [review]
Couple of curious questions

File: .github/actions/integration/entrypoint.sh (hunk 2 above)
Comment (jfalcou) [reply .github/actions/integration/entrypoint.sh:5]: Recent got update regarding to security require this in our situation

**Pull request merged**